tests/gold_tests/null_transform/null_transform.test.py

```diff
@@ -0,0 +1,73 @@
+
+'''
+'''
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import os
+Test.Summary = '''
+Test a basic null transform plugin
+'''
+
+# Need Curl
+Test.SkipUnless(
+    Condition.HasProgram("curl", "curl needs to be installed on system for this test to work")
+)
+Test.ContinueOnFail = True
+
+# Define default ATS
+ts = Test.MakeATSProcess("ts")
+server = Test.MakeOriginServer("server")
+
+Test.testName = ""
+request_header = {"headers": "GET / HTTP/1.1\r\nHost: www.example.com\r\n\r\n",
+                  "timestamp": "1469733493.993",
+                  "body":""
+                  }
+# Expected response from origin server
+response_header = {"headers": "HTTP/1.1 200 OK\r\nConnection: close\r\n\r\n",
+                   "timestamp": "1469733493.993",
+
+                   "body":"This is expected response."}
+
+# Add response the server dictionary
+server.addResponse("sessionfile.log", request_header, response_header)
+ts.Disk.records_config.update({
+    'proxy.config.diags.debug.enabled': 1,
+    'proxy.config.diags.debug.tags': 'null-transform'
+})
+
+# Load plugin
+ts.Disk.plugin_config.AddLine(
+    'null-transform.so'
+)
```

Comment:
If we are having trouble running the test due to missing the plugin, we might want to add a call to `Test#prepare_plugin`. It will build the plugin and install it to the correct place for the test.

Example: https://github.com/apache/trafficserver/blob/b4128c56dd5668858cc0c606c0e899d444aaca79/tests/gold_tests/body_factory/http204_response_plugin.test.py#L42